hi-ml-histopathology/src/histopathology/configs/classification/BaseMIL.py

```diff
@@ -59,31 +63,22 @@ class BaseMIL(LightningContainer):
 
     # Data module parameters:
     batch_size: int = param.Integer(16, bounds=(1, None), doc="Number of slides to load per batch.")
-    max_bag_size: int = param.Integer(1000, bounds=(0, None),
-                                      doc="Upper bound on number of tiles in each loaded bag during training stage. "
-                                          "If 0 (default), will return all samples in each bag. "
-                                          "If > 0, bags larger than `max_bag_size` will yield "
-                                          "random subsets of instances.")
-    max_bag_size_inf: int = param.Integer(0, bounds=(0, None),
-                                          doc="Upper bound on number of tiles in each loaded bag during "
-                                          "validation and test stages."
-                                          "If 0 (default), will return all samples in each bag. "
-                                          "If > 0 , bags larger than `max_bag_size_inf` will yield "
-                                          "random subsets of instances.")
-    cache_mode: CacheMode = param.ClassSelector(default=CacheMode.MEMORY, class_=CacheMode,
-                                                doc="The type of caching to perform: "
-                                                    "'memory' (default), 'disk', or 'none'.")
-    precache_location: CacheLocation = param.ClassSelector(default=CacheLocation.NONE, class_=CacheLocation,
-                                                           doc="Whether to pre-cache the entire transformed dataset "
-                                                               "upfront and save it to disk and if re-load in cpu or "
-                                                               "gpu. Options: `none` (default),`cpu`, `gpu`")
     encoding_chunk_size: int = param.Integer(0, doc="If > 0 performs encoding in chunks, by loading"
                                                     "enconding_chunk_size tiles per chunk")
     # local_dataset (used as data module root_path) is declared in DatasetParams superclass
 
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self.best_checkpoint_filename = "checkpoint_max_val_auroc"
+        self.best_checkpoint_filename_with_suffix = self.best_checkpoint_filename + ".ckpt"
+        self.checkpoint_folder_path = "outputs/checkpoints/"
+        self.callbacks = self.get_monitoring_callbacks(monitor="val/auroc", mode="max")
```

Comment:
I thought that this might be useful in the future in case a child class needs some extra callbacks, or do you think that will probably not be the case?

Reply:
If a child class wants additional callbacks, can't they just do the following?
```python
def get_callbacks():
  extra_callbacks = [...]
  return super().get_callbacks() + extra_callbacks
```

Reply:
yes, agree! I wasn't sure about the difference between self.callbacks and self.get_callbacks() 
because in the previous implementation self.callbacks was ModelCheckpointCallback only and self.get_callbacks() returns a list of callbacks including super().get_callbacks().
Can you confirm that there is no need to set self.callbacks as ModelCheckpointCallback only? in the latest update I set self.callbacks = self.get_callbacks()

Reply:
Yes, I believe it was only ever used inside `BaseMIL.get_callbacks()`. I'm in favour of removing any unnecessary attributes :)